src/libraries/Microsoft.Extensions.Configuration.Binder/README.md

```diff
@@ -89,3 +89,42 @@ You can include a configuration file using a code like this in your `.csproj` fi
   </Content>
 </ItemGroup>
 ```
+
+## Contributing changes to the source generator
+
+Along with the binder assembly that uses reflection, we ship a source generator assembly that provides a reflection and AOT compatible implementation. It works by intercepting regular binding calls and redirecting them to new routines that execute strongly typed binding logic.
+
+### Worfklow
+
+- [Build the .NET libraries and runtime assemblies](https://github.com/dotnet/runtime/tree/7ed33d80c92fa0b7ae740df60a460e984f2f442b#how-can-i-contribute)
+- Make the generator changes in the `gen\` directory
+- Validate the changes by running the source generator tests in `tests\SourceGeneratorTests`
+
+#### Testing
+
+Here's a general command for running the the generator tests. They assume that you're making the changes in a Windows OS, and running the commands from the `tests\SourceGeneratorTests\` directory.
+
+```ps
+dotnet build /t:rebuild;test
+```
+
+If applicable, add new tests to validate your contribution. See [this documentation](https://github.com/dotnet/runtime/blob/7ed33d80c92fa0b7ae740df60a460e984f2f442b/docs/workflow/README.md#full-instructions-on-building-and-testing-the-runtime-repo) for details.
+
+##### Stale generator bits
+
+Sometimes the SDK uses stale bits of the generator. This can lead to unexpected test behavior or failures. Killing `dotnet.exe` processes typically solves the issue, e.g. with `taskkill /F /IM dotnet.exe /T`.
+
+##### Updating baselines
+
+Some contributions might change the logic emitted by the generator. We maintain baseline [source files](https://github.com/dotnet/runtime/tree/e3e9758a10870a8f99a93a25e54ab2837d3abefc/src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines) to track the code emitted to handle some core binding scenarios.
+
+If the emitted code changes, these tests will fail locally and in continuous integration checks (for PRs) changes. You would need to update the baseline source files, manually or by using the following test command (PowerShell):
```

Comment:
>You would need to update the baseline source files, manually [](http://example.com/codeflow?start=115&length=60)

Is there a pointer where see the new emitted code? or will be displayed in the console?